bundles/core/org.openhab.core.library/src/main/java/org/openhab/core/library/items/LocationItem.java

```diff
@@ -0,0 +1,73 @@
+/**
+ * Copyright (c) 2010-2015, openHAB.org and others.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.core.library.items;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import org.openhab.core.items.GenericItem;
+import org.openhab.core.library.types.DecimalType;
+import org.openhab.core.library.types.PointType;
+import org.openhab.core.types.Command;
+import org.openhab.core.types.State;
+import org.openhab.core.types.UnDefType;
+
+
+/**
+ * This item identifies a telephone call by its origin and destination.
+ * 
+ * @author Gaël L'hopital
+ * @since 1.7.0
+ */
+public class LocationItem extends GenericItem {
+	private static List<Class<? extends State>> acceptedDataTypes = new ArrayList<Class<? extends State>>();
+	private static List<Class<? extends Command>> acceptedCommandTypes = new ArrayList<Class<? extends Command>>();
+	
+	static {
+		acceptedDataTypes.add(PointType.class);
+		acceptedDataTypes.add(UnDefType.class);
+	}
+	
+	public LocationItem(String name) {
+		super(name);
+	}
+
+	public List<Class<? extends State>> getAcceptedDataTypes() {
+		return acceptedDataTypes;
+	}
+
+	public List<Class<? extends Command>> getAcceptedCommandTypes() {
+		return acceptedCommandTypes;
+	}
+	
+	/**
+	 * Compute the distance with another Coordinate type,

```

Comment:
Code coming from here : http://stackoverflow.com/questions/837872/calculate-distance-in-meters-when-you-know-longitude-and-latitude-in-java
Bug also wondering wether it would be better to use some kind of dedicated library like this one : http://www.gavaghan.org/blog/free-source-code/geodesy-library-vincentys-formula-java/